src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -728,25 +733,6 @@ private void customLeaveToken(DetailAST ast, Deque<VariableDesc> variablesStack)
         logViolations(ast, variablesStack);
     }
 
-    /**
-     * Whether an ident with parent node of type {@link TokenTypes#METHOD_REF}
-     * should be checked or not.
-     *
-     * @param identAst identAst
-     * @return true if an ident with parent node of type {@link TokenTypes#METHOD_REF}
-     *         should be checked or if the parent type is not {@link TokenTypes#METHOD_REF}
-     * @noinspection SimplifiableIfStatement
-     */
-    public static boolean shouldCheckIdentWithMethodRefParent(DetailAST identAst) {
```

Comment:
Name of method was misleading; this method checked ASTs with many different parents such as type declarations and object initialization. I have removed the method entirely and just used two booleans in `visitIdentToken` above.